aws/resource_aws_wafregional_geo_match_set_test.go

```diff
@@ -0,0 +1,325 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSWafRegionalGeoMatchSet_basic(t *testing.T) {
+	var v waf.GeoMatchSet
+	geoMatchSet := fmt.Sprintf("tfacc-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalGeoMatchSetDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSWafRegionalGeoMatchSetConfig(geoMatchSet),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalGeoMatchSetExists("aws_wafregional_geo_match_set.test", &v),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_geo_match_set.test", "name", geoMatchSet),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_geo_match_set.test", "geo_match_constraint.#", "2"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_geo_match_set.test", "geo_match_constraint.384465307.type", "Country"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_geo_match_set.test", "geo_match_constraint.384465307.value", "US"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_geo_match_set.test", "geo_match_constraint.1991628426.type", "Country"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_geo_match_set.test", "geo_match_constraint.1991628426.value", "CA"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSWafRegionalGeoMatchSet_changeNameForceNew(t *testing.T) {
+	var before, after waf.GeoMatchSet
+	geoMatchSet := fmt.Sprintf("tfacc-%s", acctest.RandString(5))
+	geoMatchSetNewName := fmt.Sprintf("geoMatchSetNewName-%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalGeoMatchSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSWafRegionalGeoMatchSetConfig(geoMatchSet),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalGeoMatchSetExists("aws_wafregional_geo_match_set.test", &before),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_geo_match_set.test", "name", geoMatchSet),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_geo_match_set.test", "geo_match_constraint.#", "2"),
+				),
+			},
+			{
+				Config: testAccAWSWafRegionalGeoMatchSetConfigChangeName(geoMatchSetNewName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalGeoMatchSetExists("aws_wafregional_geo_match_set.test", &after),
```

Comment:
I'd expect an acceptance test here that compares `&before` to `&after` (via ID or however necessary)